src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -2,59 +2,158 @@
+
+	/* Politics */
+	national_value { nullptr },
 	government_type { nullptr },
+	last_election {},
+	ruling_party { nullptr },
+	upper_house { &ideology_keys },
+	reforms {},
+	suppression_points { 0 },
+	infamy { 0 },
 	plurality { 0 },
-	national_value { nullptr },
-	civilised { false },
+	revanchism { 0 },
+
+	/* Population */
+	primary_culture { nullptr },
+	accepted_cultures {},
+	religion { nullptr },
+	total_population { 0 },
+	national_consciousness { 0 },
+	national_militancy { 0 },
+	pop_type_distribution { &pop_type_keys },
+	national_focus_capacity { 0 },
+
+	/* Trade */
+
+	/* Diplomacy */
+	total_rank { 0 },
 	prestige { 0 },
-	upper_house { &ideology_keys },
-	technologies { &technology_keys },
-	inventions { &invention_keys } {}
+	prestige_rank { 0 },
+	industrial_power { 0 },
+	industrial_rank { 0 },
+	military_power { 0 },
+	military_rank { 0 },
+	diplomatic_points { 0 },
+
+	/* Military */
+	generals {},
+	admirals {},
+	armies {},
+	navies {},
+	regiment_count { 0 },
+	mobilisation_regiment_potential { 0 },
+	ship_count { 0 },
+	total_consumed_ship_supply { 0 },
+	max_ship_supply { 0 },
+	leadership_points { 0 },
+	war_exhaustion { 0 } {}
 
 std::string_view CountryInstance::get_identifier() const {
-	return country_definition != nullptr ? country_definition->get_identifier() : "NULL";
+	return country_definition->get_identifier();
 }
 
-bool CountryInstance::add_accepted_culture(Culture const* new_accepted_culture) {
-	if (std::find(accepted_cultures.begin(), accepted_cultures.end(), new_accepted_culture) != accepted_cultures.end()) {
-		Logger::warning(
-			"Attempted to add accepted culture ", new_accepted_culture->get_identifier(), " to country ",
-			country_definition->get_identifier(), ": already present!"
+bool CountryInstance::set_country_flag(std::string_view flag) {
+	if (flag.empty()) {
+		Logger::error("Attempted to set empty country flag for country ", get_identifier());
+		return false;
+	}
+	if (!country_flags.emplace(flag).second) {
+		Logger::error(
+			"Attempted to set country flag \"", flag, "\" for country ", get_identifier(), ": already set!"
 		);
 		return false;
 	}
-	accepted_cultures.push_back(new_accepted_culture);
 	return true;
 }
 
-bool CountryInstance::remove_accepted_culture(Culture const* culture_to_remove) {
-	auto existing_entry = std::find(accepted_cultures.begin(), accepted_cultures.end(), culture_to_remove);
-	if (existing_entry == accepted_cultures.end()) {
-		Logger::warning(
-			"Attempted to remove accepted culture ", culture_to_remove->get_identifier(), " from country ",
-			country_definition->get_identifier(), ": not present!"
+bool CountryInstance::clear_country_flag(std::string_view flag) {
+	if (flag.empty()) {
+		Logger::error("Attempted to clear empty country flag from country ", get_identifier());
+		return false;
+	}
+	if (country_flags.erase(flag) == 0) {
+		Logger::error(
+			"Attempted to clear country flag \"", flag, "\" from country ", get_identifier(), ": not set!"
```

Comment:
Clearing a flag that isn't in the set is not an error.

Reply:
Same as above